src/xcode/ENA/ENA/Source/Scenes/TestResultAvailable/TestResultAvailableViewModel.swift

```diff
@@ -4,30 +4,50 @@
 
 import Foundation
 import UIKit
+import Combine
 
-struct TestResultAvailableViewModel {
-
+final class TestResultAvailableViewModel {
+	
 	// MARK: - Init
-
+	
 	init(
-		_ store: Store,
+		exposureSubmissionService: ExposureSubmissionService,
 		didTapConsentCell: @escaping () -> Void,
 		didTapPrimaryFooterButton: @escaping () -> Void,
 		presentDismissAlert: @escaping () -> Void
 	) {
-		self.store = store
+		self.exposureSubmissionService = exposureSubmissionService
 		self.didTapConsentCell = didTapConsentCell
 		self.didTapPrimaryFooterButton = didTapPrimaryFooterButton
 		self.presentDismissAlert = presentDismissAlert
+		
+		exposureSubmissionService.isSubmissionConsentGivenPublisher.sink { [weak self] consentGranted in
+			self?.dynamicTableViewModel = self?.createDynamicTableViewModel(consentGranted)
+		}.store(in: &cancellables)
 	}
-
+	
 	// MARK: - Internal
-
+	
 	let didTapPrimaryFooterButton: () -> Void
 	let presentDismissAlert: () -> Void
-
-	var dynamicTableViewModel: DynamicTableViewModel {
-		DynamicTableViewModel([
+	
+	@Published var dynamicTableViewModel: DynamicTableViewModel?
```

Comment:
You could initialize this var with an empty DynamicTableViewModel to get rid of the optionality